### PR TITLE
prune-junit-xml cleanup

### PR DESCRIPTION
**Pull request by BenTheElder**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:
- simplifies logic ... NArgs() returns len(Args())
- adds OWNERS for this tool, I added existing contributors and some active SIG Testing Leads and the sig/testing label

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

/hold
/assign @dims
cc @aojea @pohly @liggitt

### CODE REVIEW

File: cmd/prune-junit-xml/prunexml.go [hunk 1]
```diff
@@ -29,33 +29,30 @@ import (
 func main() {
 	maxTextSize := flag.Int("max-text-size", 1, "maximum size of attribute or text (in MB)")
 	flag.Parse()
+	for _, path := range flag.Args() {
```
Comment (BenTheElder) [inline cmd/prune-junit-xml/prunexml.go:32]: git diff is struggling. this is just de-indent and drop the `if flag.NArg() > 0`, `for _, path := range flag.Args()` is already a no-op if len(Args()) is 0.

Comment (BenTheElder) [reply cmd/prune-junit-xml/prunexml.go:32]: yamlfmt has the same unnecessary check,  https://github.com/kubernetes/kubernetes/pull/118066

**Comment by BenTheElder** [comment]
Hmm scheduling limits in CI
https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/118065/pull-kubernetes-verify/1658720172304764928
/retest

File: cmd/prune-junit-xml/OWNERS [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# See the OWNERS docs at https://go.k8s.io/owners
+
+reviwers:
```
Comment (pohly) [inline cmd/prune-junit-xml/OWNERS:3]: Typo:
```suggestion
reviewers:
```

**Review by pohly — CHANGES_REQUESTED** [review]

**Review by pohly — APPROVED** [review]
/lgtm

**Pull request merged**